websites/S/Stremio/presence.ts

```diff
@@ -3,172 +3,404 @@ const presence = new Presence({
+					presenceData.details = `Discovering ${type ?? "content"}${
+						type === "series" ? "" : "s"
+					}`;
+					presenceData.state = `${category ?? "All"}${
+						genre ? ` | ${genre}` : ""
+					}`;
+
+					break;
+				}
+				case "library": {
+					const type = document.querySelector(
+						appVersion === AppVersion.V4
+							? "[class='ng-binding ng-scope selected']"
+							: "div[class*='selectable-inputs-container'] > div > div"
+					)?.textContent;
+
+					presenceData.details = "Library";
+					presenceData.state = type ?? "All";
+					presenceData.buttons = [
+						{
+							label: "View library",
+							url: href,
+						},
+					];
+					break;
+				}
+				case "calendar":
+					presenceData.details = "Calendar";
+					presenceData.buttons = [
+						{
+							label: "View calendar",
+							url: href,
+						},
+					];
+					break;
+				case "search":
+					presenceData.details = "Search";
+					break;
+				case "player": {
+					if (video === null) break;
+
+					presenceData.endTimestamp = video.endTimestamp;
+					delete presenceData.startTimestamp;
+
+					if (
+						(appVersion === AppVersion.V4
+							? document
+									.querySelector("#loading-logo")
+									.className.includes("flashing")
+							: !!document.querySelector(
+									"div[class*='buffering-loader-container']"
+							  )) ||
+						video.isPaused
+					) {
+						presenceData.smallImageKey = Assets.Pause;
+						presenceData.smallImageText = "Player is paused";
+						presenceData.state = "Paused";
+					} else {
+						presenceData.smallImageKey = Assets.Play;
+						presenceData.smallImageText = "Player is playing";
+						presenceData.state = "Watching";
+					}
+
+					let metaUrl: string, title: string;
+
+					if (appVersion === AppVersion.V4) {
+						title = document
+							.querySelector("head > title")
+							?.textContent?.replace("Stremio -", "")
+							?.trim();
+						metaUrl = href
+							.substring(0, href.lastIndexOf("/"))
+							.replace("player", "detail");
+						presenceData.largeImageKey =
+							document
+								.querySelector("#loading-logo")
+								?.getAttribute("data-image") ?? Assets.Logo;
+					} else {
+						const playerState = await _eval(
+							"core.transport.getState('player')"
+						);
+						if (playerState.metaItem.type.toLowerCase() === "ready") {
+							// eslint-disable-next-line prefer-destructuring
+							const content = playerState.metaItem.content;
+							// eslint-disable-next-line prefer-destructuring
+							title = playerState.title;
+							metaUrl = `${window.location.origin}/#/detail/${content.type}/${content.id}`;
+							if (content.type === "series")
+								metaUrl += `/${content.id}:${playerState.seriesInfo.season}:${playerState.seriesInfo.episode}`;
+							presenceData.largeImageKey = content.logo ?? Assets.Logo;
+						}
+					}
+
+					presenceData.details = title ?? "Player";
+					if (metaUrl) {
+						presenceData.buttons = [
+							{
+								label: "Watch",
```

Comment:
```suggestion
								label: "Watch Video",
```

Reply:
I intentionally changed it to `watch` because the link doesn't go directly to a video. It goes to the meta page instead. See the PR description for a detailed explanation. I didn't want to change it to `view metadata` either because that doesn't make sense in that context, so just `watch` felt like the perfect middle-ground.